query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLInstantQuery.java

```diff
@@ -0,0 +1,19 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import java.time.Instant;
+import java.util.List;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Singular;
+
+@Getter
+@Builder
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+class PromQLInstantQuery {
+  @NonNull @Singular private List<String> queries;
+
+  @NonNull private Instant evalTime;
```

Comment:
for instant query also we need start and end time?

Reply:
No for Instant query, just evaluation time is required, and kept the name based on the argument of PromQL API - https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries

As describe here - https://github.com/hypertrace/query-service/pull/110/#issue-1065988936, the time filter duration (endTime - startTime) will go into duration query expression of PromQL`[]`, and your endTime QS's time filter will be `evalTime`

Repeating the above example: 
```
sum by (backend_id) 
(sum_over_time(num_calls{tenant_id="__default"}[3600000ms])) 
@ 1631181508474

sum by (backend_id) 
(sum_over_time(error_count{tenant_id="__default"'}[3600000ms])) 
@ 1631181508474
```

As in the above example, see `[3600000ms]` that's your time filter of QS's aggregation query. And `@ 1631181508474` is your QS's time filter endTime, is `evalTime` for promQL query.

```
filter {
  childFilter {
    childFilter {
      lhs {
        columnIdentifier {
          columnName: "BACKEND.startTime"
        }
      }
      operator: GE
      rhs {
        literal {
          value {
            valueType: LONG
            long: 1631177908474
          }
        }
      }
    }
    childFilter {
      lhs {
        columnIdentifier {
          columnName: "BACKEND.startTime"
        }
      }
      operator: LT
      rhs {
        literal {
          value {
            valueType: LONG
            long: 1631181508474
          }
        }
      }
    }
  }
}
```

Reply:
Ok, maybe you can add a comment on what eval time is?

Reply:
Added comments.

Reply:
Ah, I asked this on #111 - there's quite a bit of overlap. So an instant query still applies to a range of time, but one argument is embedded in the query string and one is in a query param?

Reply:
yes, that is correct